pkg/telemetrylogs/statement_builder.go

```diff
@@ -478,11 +524,33 @@ func (b *logQueryStatementBuilder) buildScalarQuery(
 
 	var allGroupByArgs []any
 
+	// Collect array join info for body JSON fields
+	var arrayJoinClauses []string
+
 	for _, gb := range query.GroupBy {
-		expr, args, err := querybuilder.CollisionHandledFinalExpr(ctx, &gb.TelemetryFieldKey, b.fm, b.cb, keys, telemetrytypes.FieldDataTypeString, b.jsonBodyPrefix, b.jsonKeyToKey)
-		if err != nil {
-			return nil, err
+		var expr string
+		var args []any
+		var err error
+
+		// For body JSON fields with feature flag enabled, use array join logic
+		if strings.HasPrefix(gb.TelemetryFieldKey.Name, BodyJSONStringSearchPrefix) && constants.BodyV2QueryEnabled {
+			// Build array join info for this field
```

Comment:
for these will it be clear to create v1/ v2 internal functions and then call whatever is required, even though code will be duplicated.

That way removing the flag will be cleaner.